processor/consent.go

```diff
@@ -1,59 +1,131 @@
 package processor
 
 import (
+	"fmt"
+
 	"github.com/samber/lo"
 
 	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 	"github.com/rudderlabs/rudder-server/utils/types"
 )
 
-// filterDestinations filters destinations based on consent categories, supports oneTrustCookieCategories and ketchConsentPurposes
-func (proc *Handle) filterDestinations(event types.SingularEventT, destinations []backendconfig.DestinationT) []backendconfig.DestinationT {
-	// If there are no deniedConsentIds, then return all destinations
-	deniedCategories := deniedConsentCategories(event)
-	if len(deniedCategories) == 0 {
+type ConsentManagementInfo struct {
+	DeniedConsentIDs   []string `json:"deniedConsentIds"`
+	AllowedConsentIDs  []string `json:"allowedConsentIds"` // Not used currently but added for future use
+	Provider           string   `json:"provider"`
+	ResolutionStrategy string   `json:"resolutionStrategy"`
+}
+
+type GenericConsentManagementProviderData struct {
+	ResolutionStrategy string
+	Consents           []string
+}
+
+type GenericConsentsConfig struct {
+	Consent string `json:"consent"`
+}
+
+type GenericConsentManagementProviderConfig struct {
+	Provider           string                  `json:"provider"`
+	ResolutionStrategy string                  `json:"resolutionStrategy"`
+	Consents           []GenericConsentsConfig `json:"consents"`
+}
+
+/*
+Filters and returns destinations based on the consents configured for the destination and the user consents present in the event.
+
+Supports legacy and generic consent management.
+*/
+func (proc *Handle) getConsentFilteredDestinations(event types.SingularEventT, destinations []backendconfig.DestinationT) []backendconfig.DestinationT {
+	// If the event does not have denied consent IDs, do not filter any destinations
+	consentManagementInfo, err := getConsentManagementInfo(event)
+	if err != nil {
+		// Log the error for debugging purposes
+		proc.logger.Error(err)
```

Comment:
We can use structured logging here. Same for the other logging as well.
```suggestion
		proc.logger.Errorw("failed to get consent management info", "error", err.Error())
```